beacon-chain/p2p/service.go

```diff
@@ -649,3 +654,20 @@ func logExternalDNSAddr(id peer.ID, addr string, port uint) {
 		).Info("Node started external p2p server")
 	}
 }
+
+// Attempt to dial an address to verify its connectivity
+func verifyExternalIPAddr(addr string, port uint) {
```

Comment:
can you also add verification for the udp port provided ?

Reply:
added.

Reply:
That is not what I meant, we have 2 different ports for both tcp and udp. So we also need to ensure they are publically reachable for both. You can access the set udp port in the service through `s.cfg.UDPPort`. Use that to also check the validity of it.

Reply:
Hmm, unless I'm missing something, it'll be quite challenging to ensure that a PORT is open for UDP communications.
The updated current implementation (leveraging ListenUDP) will notify if an IP address isn't reachable, but I don't think, due to the nature of UDP (connectionless), we can detect if a specific port is open or not.

Reply:
Not sure I understand, you can `Dial` the relevant IP + port to validate that there is a listening port running there. This works for both `TCP` and `UDP` . We have listening ports running for both protocols.